src/storage/admin/RebuildEdgeIndexProcessor.cpp

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "storage/admin/RebuildEdgeIndexProcessor.h"
+
+namespace nebula {
+namespace storage {
+
+void RebuildEdgeIndexProcessor::process(const cpp2::RebuildIndexRequest& req) {
+    CHECK_NOTNULL(kvstore_);
+    auto space = req.get_space_id();
+    auto parts = req.get_parts();
+    auto indexID = req.get_index_id();
+    auto itemRet = indexMan_->getEdgeIndex(space, indexID);
+    if (!itemRet.ok()) {
+        cpp2::ResultCode thriftRet;
+        thriftRet.set_code(cpp2::ErrorCode::E_INDEX_NOT_FOUND);
+        codes_.emplace_back(thriftRet);
+        onFinished();
+        return;
+    }
+    auto item = itemRet.value();
+    auto edgeType = item->get_schema_id().get_edge_type();
+    LOG(INFO) << "Rebuild Edge Index Space " << space << " Edge Type " << edgeType
+              << " Edge Index " << indexID;
+
+    for (PartitionID part : parts) {
+        // firstly remove the discard index
+        auto indexPrefix = NebulaKeyUtils::indexPrefix(space, indexID);
+        doRemovePrefix(space, part, std::move(indexPrefix));
```

Comment:
I'm confused about asyncRemovePrefix. WDYT？

Reply:
if the index have built, you should remove firstly.

Reply:
> if the index have built, you should remove it firstly.

Sure, I means is async remove reliable?  Is sync remove better?